src/RevokePassportTokens.php

```diff
@@ -28,15 +24,38 @@ public function handle(TokenRepository $tokenRepository)
             return;
         }
 
+        $user = $this->option('user');
+        $client = $this->option('client');
+
         $query = Passport::token()->newQuery()
             ->where('revoked', false)
             ->where('expires_at', '>=', Carbon::now());
 
-        if ($user = $this->option('user')) {
-            $query->where('user_id', $user);
-            $this->info("Revoking all tokens for user $user...");
-        } else {
-            $this->info('Revoking all active tokens...');
+        switch (true) {
```

Comment:
a switch on `true` is generally bad practice – I think it might be better if we could turn this into just a couple if statements. That would also allow reducing repetition. e.g

```
if ($user) {
    $query->where('user_id', $user);
}

if ($client) {
    $query->where('client_id', $client);
}

if (! ($client || $user)) {
    if (!$this->confirm('You did not provide any user, client or token. All Passport tokens will be revoked. Continue?') {
        return;
    }
     $this->info('Revoking all active tokens...');
}
```

And of course also handling building up the info string, easily done with e.g
```
$message = 'Revoken all tokens for ';
$constraints = [];

if ($user) {
    $constraints[] = "user $user";
}

$message .= implode(' and ', $constraints);
```

Reply:
Saying something is a bad practice without an argument is not really productive. Neither is nitpicking on every detail, especially when the script is as simple as it gets and not covered by tests in the first place.  

A switch on `true` can make something much more readable and maintainable than some if-else spaghetti. Usually it also reduced code duplication. In this case, there is some minor duplication, but for the use-case, it is very acceptable. You do not need a bulldozer to remove a brick.

I'll let you handle the PR from here on so it adheres to your standards.

Reply:
My apologies if I came across as harsh, I was mostly trying to avoid repetition where it wasn't necessary. I'm fine with a switch on true if you think that's the best alternative, I just found it a bit confusing to read a switch that dealt with multiple conditions (client and user). If you think this is the most clean solution, let's roll with that

Reply:
After checking out this latest version I see very little concrete benefit in improving and/or tweaking the implementation any further from here, and therefor I'll be merging this PR as it is. 

We'll be adding in some tests and a CI pipeline setup for the project _probably_ in the upcoming weeks, which should make updating more comfortable in the future. 😎

Reply:
Haha, wasn't expecting you to hop online from a holiday 😉 👌